x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -97,11 +97,11 @@ func (k Keeper) OptInTopNValidators(ctx sdk.Context, chainID string, bondedValid
 }
 
 // ComputeMinPowerToOptIn returns the minimum power needed for a validator (from the bonded validators)
-// to belong to the `topN` validators. `chainID` is only used for logging purposes.
+// to belong to the `topN` validators for a Top N chain.
 func (k Keeper) ComputeMinPowerToOptIn(ctx sdk.Context, chainID string, bondedValidators []stakingtypes.Validator, topN uint32) (int64, error) {
-	if topN == 0 || topN > 100 {
-		return 0, fmt.Errorf("trying to compute minimum power with an incorrect topN value (%d)."+
-			"topN has to be between (0, 100]", topN)
+	if topN < 50 || topN > 100 {
```

Comment:
Based on the "No validation for TopN property" audit's comment. We did not add an additional check in `HandleOptOut` because `HandleOptOut` calls `ComputeMinPowerToOptIn`, so if `topN` is not correctly set, `HandleOptOut` would fail as well.

Reply:
I'm not super convinced of this; if someone, for some as-of-yet-unforeseen reason wants to reuse the calculation, but with topN < 50, they have to change this code? (or even worse, fork it)

Not sure it's worth it, we could have something like `ComputeMinPowerToOptInUnchecked` that's just the actual logic, and then the current version of `ComputeMinPowerToOptIn` that includes the check, and then calls `ComputeMinPowerToOptInUnchecked`
But I'm not super adamant about this, leaving as is is not a big deal

Reply:
I agree. I fixed it to take any `topN` value in `(0, 100]` (as it was previously) and also added a comment. I do not think it's worth creating another method (e.g., `ComputeMinPowerToOptInUnchecked `) for this.